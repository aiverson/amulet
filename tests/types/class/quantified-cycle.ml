type t1 'a = T1
type t2 'a = T2

class c 'a begin
  val meth : 'a -> ()
  let meth _ = ()
end

instance (forall 'a. c (t1 'a)) => c (t2 'a) begin end
instance (forall 'a. c (t2 'a)) => c (t1 'a) begin end

let () = meth T1
