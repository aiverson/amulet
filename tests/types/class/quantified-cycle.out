quantified-cycle.ml[12:10 ..12:16]: error
  No instance for c (t1 'a) arising in this expression
   │ 
12 │ let () = meth T1
   │          ^^^^^^^
  • Note: This constraint can not be quantified over because it is of higher rank
